include/ruby/internal/intern/error.h

```diff
@@ -248,12 +249,27 @@ RBIMPL_SYMBOL_EXPORT_END()
         if (RB_UNLIKELY(RB_OBJ_FROZEN(frozen_obj))) { \
             rb_error_frozen_object(frozen_obj); \
         } \
+        if (RB_UNLIKELY(rb_str_chilled_p(frozen_obj))) { \
+            rb_str_modify(frozen_obj); \
+        } \
     } while (0)
 
+# ifdef __cplusplus
+extern "C" {
+# endif
+bool rb_str_chilled_p(VALUE str);
+void rb_str_modify(VALUE str);
+# ifdef __cplusplus
+}
+# endif
```

Comment:
```suggestion
RBIMPL_SYMBOL_EXPORT_BEGIN()

bool rb_str_chilled_p(VALUE str);
void rb_str_modify(VALUE str);

RBIMPL_SYMBOL_EXPORT_END()
```

Or you can move these declarations before `RBIMPL_SYMBOL_EXPORT_END()`.

Reply:
Thank you, I was looking for a clean way to do this.